tests/integration/modules/test_archive.py

```diff
@@ -2,44 +2,46 @@
 """
 Tests for the archive state
 """
-# Import python libs
 from __future__ import absolute_import, print_function, unicode_literals
 
 import os
 import shutil
 import textwrap
 
-# Import salt libs
+import pytest
 import salt.utils.files
 import salt.utils.path
 import salt.utils.platform
 import salt.utils.stringutils
-
-# Import 3rd party libs
 from salt.ext import six
 from tests.support.case import ModuleCase
 from tests.support.helpers import destructiveTest
-
-# Import Salt Testing libs
 from tests.support.runtests import RUNTIME_VARS
 from tests.support.unit import skipIf
 
 try:
-    import zipfile  # pylint: disable=W0611
+    import zipfile  # pylint: disable=unused-import
 
     HAS_ZIPFILE = True
 except ImportError:
     HAS_ZIPFILE = False
 
 
 @destructiveTest
+@pytest.mark.windows_whitelisted
 class ArchiveTest(ModuleCase):
     """
     Validate the archive module
     """
 
-    # Base path used for test artifacts
-    base_path = os.path.join(RUNTIME_VARS.TMP, "modules", "archive")
+    @classmethod
+    def setUpClass(cls):
+        # Base path used for test artifacts
+        cls.base_path = os.path.join(RUNTIME_VARS.TMP, "modules", "archive")
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.base_path = None
```

Comment:
Why does this need to be setup or torn down?

Reply:
Since it's a class attribute, it's less "garbage" to keep in memory until the class is, eventually, GC'ed